File: Front/ProEvents-App/src/app/models/Event.ts
import { EventSpeaker } from "./EventSpeaker";
import { Lote } from "./Lote";
import { SocialMedia } from "./SocialMedia";

export interface Event {
  id: number;
  place: string;
  eventDate?: Date;
  theme: string;
  numberOfPerson: number;
  imageUrl: string;
  phone: string;
  email: string;
  lotes: Lote[];
  socialMedias: SocialMedia[];
  eventSpeakers: EventSpeaker[];
  userId: number;
}

File: Back/src/ProEvents.Domain/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using ProEvents.Domain.Identity;

namespace ProEvents.Domain
{
    public class Event
    {
        public int Id { get; set; }
        [Required]
        public string Place { get; set; }
        [Required]
        public DateTime? EventDate { get; set; }
        [Required]
        public string Theme { get; set; }
        [Required]
        public int NumberOfPerson { get; set; }
        public string ImageUrl { get; set; }
        [Required]
        public string phone { get; set; }
        [Required]
        public string Email { get; set; }
        public IEnumerable<Lote> Lotes { get; set; }
        public IEnumerable<SocialMedia> SocialMedias { get; set; }
        public IEnumerable<EventSpeaker> EventSpeakers { get; set; }
        public User User { get; set; }
        public int UserId { get; set; }
    }
}
